WeakAuras/Profiling.lua

```diff
@@ -360,3 +366,160 @@ function WeakAuras.PrintProfile()
+  bar.fg:ClearAllPoints()
+  bar.fg:SetPoint("TOPLEFT", bar)
+  bar.fg:SetSize(self.width / 100 * pct, self.barHeight)
+  bar.txtName:SetText(("%s (%.2fms)"):format(name, map.elapsed))
+  bar.txtPct:SetText(("%.2f%%"):format(pct))
+  bar:Show()
+  return bar
+end
+
+function RealTimeProfilingWindow:RefreshBars()
+  if not profileData.systems.time then
+    return
+  end
+
+  local total = TotalProfileTime(WeakAuras.profileData.auras)
+  for i, k in ipairs(SortProfileMap(WeakAuras.profileData.auras)) do
+    if (k ~= "time" and k ~= "wa") then
+      local bar = self:AddBar(i, k, profileData.auras[k], total)
+      if bar then
+        if i % 2 == 0 then
+          bar.fg:SetColorTexture(1, 1, 1, 0.7)
+          bar.bg:SetColorTexture(0, 0, 0, 0.2)
+        else
+          bar.fg:SetColorTexture(1, 1, 1, 0.9)
+          bar.bg:SetColorTexture(0, 0, 0, 0.4)
+        end
+      end
+    end
+  end
+  local timespent = debugprofilestop() - profileData.systems.time.start
+  self.statsFrameText:SetText(("Time in WeakAuras: %.2f%%"):format(100 * profileData.systems.wa.elapsed / timespent))
+end
+
+function RealTimeProfilingWindow:Start()
+  self:ClearAllPoints()
+  self:SetSize(self.width, self.height)
+  self:SetClampedToScreen(true)
+  self:SetPoint("TOPLEFT", UIParent, "TOPLEFT")
+  self:Show()
+  if not self.titleFrame then
+    local titleFrame = CreateFrame("Frame", nil, self)
+    self.titleFrame = titleFrame
+    titleFrame:SetSize(self.width, self.titleHeight)
+    titleFrame:SetPoint("TOPLEFT", self)
+    titleFrame:Show()
+    local titleText = self.titleFrame:CreateFontString(nil, "OVERLAY", "GameFontNormal")
+    self.titleFrameText = titleText
+    titleText:SetText(L["WeakAuras Profiling"])
+    titleText:SetPoint("CENTER", self.titleFrame)
+  end
+  if not self.bg then
+    local bg = self:CreateTexture(nil, "BACKGROUND")
+    self.bg = bg
+    bg:SetTexture(texture)
+    bg:SetAllPoints()
+    bg:Show()
+  end
+  if not self.statsFrame then
+    local statsFrame = CreateFrame("Frame", nil, self)
+    self.statsFrame = statsFrame
+    statsFrame:SetSize(self.width, self.titleHeight)
+    statsFrame:SetPoint("TOPLEFT", self, "BOTTOMLEFT")
+    statsFrame:Show()
+    local statsFrameText = self.statsFrame:CreateFontString(nil, "OVERLAY", "GameFontNormal")
+    self.statsFrameText = statsFrameText
+    statsFrameText:SetPoint("LEFT", self.statsFrame)
+  end
+  self:SetScript("OnMouseDown", function(self, button)
+    if button == "LeftButton" and not self.is_moving then
+      self:StartMoving()
+      self.is_moving = true
+    elseif button == "RightButton" then
+      self:Stop()
+    end
+  end)
+
+  self:SetScript("OnMouseUp", function(self, button)
+    if button == "LeftButton" and self.is_moving then
+      self:StopMovingOrSizing()
+      self.is_moving = nil
+    end
+  end)
+
+  self:SetScript("OnUpdate", self.RefreshBars)
+end
+
+function RealTimeProfilingWindow:Stop()
+  self:Hide()
+  self:SetScript("OnUpdate", nil)
```

Comment:
This is implicitly covered by Hide(), since OnUpdate isn’t dispatched to hidden frames.

Reply:
I didn't know 👍